slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -161,6 +161,10 @@ class $name(_tableTag: Tag) extends profile.api.Table[$elementType](_tableTag, $
       def columnOptionCode = {
         case ColumnOption.PrimaryKey => Some(s"O.PrimaryKey")
         case Default(value)     => Some(s"O.Default(${default.get})") // .get is safe here
+        case Default.Now        => Some("O.Default.Now")
+        case Default.CurrentDate => Some("O.Default.CurrentDate")
+        case Default.CurrentTime => Some("O.Default.CurrentTime")
+        case Default.CurrentTimestamp => Some("O.Default.CurrentTimestamp")
```

Comment:
Instead of naming 4 particular cases, what do you think about having something like `DefaultExpression("...")` that we can use to pass anything through? Like `O.DefaultExpression("NOW()")`, `O.DefaultExpression("CURRENT_DATE")`, ... 

That would allow people to call other functions as well and would require less handling of specific cases in Slick.